src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -352,12 +363,64 @@ public async Task DeletePullRequestBranchAsync(string pullRequestUri)
             }
         }
 
-        public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters)
+        /// <summary>
+        ///     Parse out the owner and repo from a repository url
+        /// </summary>
+        /// <param name="pattern">Regex Pattern</param>
+        /// <param name="prBody">Pull Request commit msg</param>
+        /// <returns>Tuple of owner and repo</returns>
+        public string ParsePullRequestBody(Regex pattern, string prBody)
+        {
+            var matches = pattern.Matches(prBody);
+            if (matches.Count == 0)
+            {
+                return string.Empty;
+            }
+            string message = "";
+            foreach (Match match in matches)
+            {
+                message += $@"
+{match.Value.ToString().Replace("*", "")}";
+            }
+            return message;
+        }
+
+        /// <summary>
+        /// Merges the dependency pull request 
+        /// </summary>
+        /// <param name="pullRequestUrl"></param>
+        /// <param name="parameters"></param>
+        /// <returns></returns>
+        public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters)
         {
             CheckForValidGitClient();
             _logger.LogInformation($"Merging pull request '{pullRequestUrl}'...");
 
-            await _gitClient.MergePullRequestAsync(pullRequestUrl, parameters ?? new MergePullRequestParameters());
+            var pr = await _gitClient.GetPullRequestAsync(pullRequestUrl);
+            var commits = await _gitClient.GetPullRequestCommitsAsync(pullRequestUrl);
+            string dependencyUpdate = ParsePullRequestBody(DependencyUpdatesPattern, pr.Description);
+            string coherencyUpdate = ParsePullRequestBody(CoherencyUpdatesPattern, pr.Description);
+            string commitMessage = $@"{pr.Title}
+{dependencyUpdate}
+
+";
+            foreach (Commit commit in commits)
+            {
+                if (!commit.Author.Equals("dotnet-maestro[bot]"))
+                {
+                    commitMessage += $@"- {commit.Message}";
```

Comment:
Are these commits some machine generated commits? Because otherwise the formatting is going to get very, very strange, since the will have their own formatting inside.  If they aren't machine generated (and I expecte they are not, with the Author check above), it would probably be better to separate them with `---` to make it a horizontal rule, possibly with the sha/author information included in a header, since that might be interesting.

Reply:
I am tracking manual commit here. So now the manual commits looks like this

https://dev.azure.com/dnceng/internal/_git/maestro-test2/commit/f92b43ece9b875feb1b9f63d3949a9e240acdbbb?refName=refs%2Fheads%2Fdarc-test%2F19129235-2e18d7fa-4c0a-4a2d-926f-ab520d84e572&tab=details